packages/web/src/io/getGeneMap.ts

```diff
@@ -40,6 +46,21 @@ export function getGeneMap(geneMapJson: GeneMapJson): Gene[] {
     const color = GENOTYPE_COLORS[(i + 1) % GENOTYPE_COLORS.length]
     const { start: begin, end } = pick(geneDataRaw, ['start', 'end'])
     const geneWithoutColor = { name, range: { begin, end } }
+
+    const length = end - begin
+
+    if (length) {
+      throw new ErrorGeneInvalidRange(
+        `GeneMap: in gene "${name}": expected gene.begin < gene.end, but got ${begin} < ${end}`,
+      )
+    }
+
+    if (!isDivisibleBy(length, 3)) {
+      throw new ErrorGeneInvalidRange(
+        `GeneMap: in gene "${name}": expected gene length to be divisible by 3, but got ${length}`,
+      )
+    }
```

Comment:
This should ensure that we won't go out of bounds when iterating gene sequences with `i += 3` increments.